lib/skywalking/tracing_context.lua

```diff
@@ -109,14 +109,21 @@ function _M.newNoOP()
     return {is_noop = true}
 end
 
-function _M.new(serviceName, serviceInstanceName)
+function _M.new(serviceName, serviceInstanceName, requestId)
     if serviceInstanceName == nil or serviceName == nil then
         return _M.newNoOP()
     end
 
+    -- use request_id as trace_id if it is present
+    local trace_id = Util.newID()
+    local segment_id = trace_id
+    if requestId ~= nil then
+        trace_id = requestId
+    end
```

Comment:
`Util.newID()` is not free so it'd better only invoke it when necessary

```suggestion
    local trace_id = requestId
    if trace_id == nil then
        trace_id = Util.newID()
    end
    local segment_id = trace_id
```

Reply:
yeah.
but segment id has to invoke it every time no matter about request id exists or not.

Reply:
Why? If request ID exists, you are going to use request ID as trace id, then newID could be skipped.

Reply:
emmm. the code is a bad readablity.
i means, trace id is from request id. but the segment id still has to generate a new one. 
i invoke the newID to generate an id for trace id and segment id at same time.
the request id cannot be segment id because the segment id conflicts when the upstream is nginx.

Reply:
I see, if it is about segment id, then yes.

Reply:
If request id is nil, the trace id is the same as segment id?

Reply:
Yes, they were the same.